staging/src/k8s.io/pod-security-admission/policy/options.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policy
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type options struct {
+	withFieldErrors bool
+}
+
+type Option func(options) options
+
+func withOptions(f func(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec, opts options) CheckResult) CheckPodFn {
+	return func(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec, opts ...Option) CheckResult {
+		var opt options
+		for _, o := range opts {
+			if o != nil {
+				opt = o(opt)
+			}
+		}
+		return f(podMetadata, podSpec, opt)
+	}
+}
+
+func WithFieldErrors() Option {
+	return func(opt options) options {
+		opt.withFieldErrors = true
+		return opt
+	}
+}
```

Comment:
This pattern looks like overkill for this usecase. Is there a reason you prefer this approach? I'd just pass the options struct directly (make it exported, with exported fields)

Reply:
Here we change the `CheckPodFn` to use a variadic argument because we want to change the signature without impacting the caller. Users can also decide whether to turn on the capability as needed, as follows:

```
checkResult := latestVersionCheck.CheckPod(&pod.ObjectMeta, &pod.Spec, policy.WithFieldErrors())
```
cc @eddycharly

Reply:
@tallclair this is to not break existing code.